src/main/java/org/folio/circulation/infrastructure/storage/PrintEventsRepository.java

```diff
@@ -32,4 +48,63 @@ public CompletableFuture<Result<Void>> create(PrintEventRequest printEventReques
     return printEventsStorageClient.post(storagePrintEventRequest).thenApply(interpreter::flatMap);
   }
 
+  public CompletableFuture<Result<MultipleRecords<Request>>> findPrintEventDetails(
+    MultipleRecords<Request> multipleRequests) {
+    log.debug("findPrintEventDetails:: parameters multipleRequests: {}",
+      () -> multipleRecordsAsString(multipleRequests));
+    return validatePrintEventFeatureFlag()
+      .thenCompose(isEnabled -> {
+        if (Boolean.TRUE.equals(isEnabled)) {
+          log.info("findPrintEventDetails:: printEvent feature is enabled for the tenant");
+          return fetchAndMapPrintEventDetails(multipleRequests);
+        } else {
+          log.info("findPrintEventDetails:: printEvent feature is disabled for the tenant");
+          return completedFuture(succeeded(multipleRequests));
+        }
+      });
+  }
+
+  private CompletableFuture<Result<MultipleRecords<Request>>> fetchAndMapPrintEventDetails(
+    MultipleRecords<Request> multipleRequests) {
+    log.debug("fetchAndMapPrintEventDetails:: parameters multipleRequests: {}",
+      () -> multipleRecordsAsString(multipleRequests));
+    var requestIds = multipleRequests.toKeys(Request::getId);
+    if (requestIds.isEmpty()) {
+      log.info("fetchAndMapPrintEventDetails:: No request id found");
+      return completedFuture(succeeded(multipleRequests));
+    }
+    return fetchPrintDetailsByRequestIds(requestIds)
+      .thenApply(printEventRecordsResult -> printEventRecordsResult
+        .next(printEventRecords -> mapPrintEventDetailsToRequest(printEventRecords, multipleRequests)));
+  }
+
+  private CompletableFuture<Boolean> validatePrintEventFeatureFlag() {
+    log.debug("validatePrintEventFeatureFlag:: Fetching and validating enablePrintLog flag from settings");
+    return circulationSettingsRepository.findBy("query=name=printEventLogFeature")
```

Comment:
Its better to provide a string constant for the printEventLogFeature name

Reply:
Done